### PR TITLE
Cleans up leaderboard endpoint and allows permission to access to admin and cd.

### PR DESCRIPTION
**Pull request by eet349**
Description
The leaderboard endpoint needed cleaning up. We also needed to allow access to the endpoint to the correct authenticated users.
What work was done?  
Created service methods that called their respective repositories. Previously the controller was calling the repositories directly, but now they call the services. Also added rules to allow the /leaderboard/leaderboard to be accessed by authenticated Club director and Admin accounts.

Why was this work done?  
To allow access to the frontend to fetch the data needed.

[x] Completed, ready to review and merge  
Checklist
[x] My code follows the style guidelines of this project  
[x] I have performed a self-review of my own code  
[] My code has been reviewed by at least one peer  
[] I have commented my code, particularly in hard-to-understand areas  
[] I have made corresponding changes to the documentation  
[x] My changes generate no new warnings  
[x] There are no merge conflicts

### CODE REVIEW

File: src/main/java/com/lambdaschool/oktafoundation/controllers/LeaderboardController.java [hunk 1]
```diff
@@ -28,33 +26,23 @@
     private ClubService clubService;
 
     @Autowired
-    private ClubRepository clubRepository;
+    ReactionService reactionService;
```
Comment (L-Steinmacher) [inline src/main/java/com/lambdaschool/oktafoundation/controllers/LeaderboardController.java:30]: Yeah Services!

File: src/main/java/com/lambdaschool/oktafoundation/controllers/LeaderboardController.java [hunk 2]
```diff
@@ -75,10 +63,11 @@
                 }
             }
             checkout = rtnList.get(checkoutIndex);
-            ClubsCheckInOutSummary finalCheckout = checkout;
-            ClubsCheckInOutSummary finalCheckin = checkin;
-            Reaction checkoutReaction =  reactionRepo.findReactionByReactionid(checkout.getReactionid()).orElseThrow(() -> new ResourceNotFoundException("reaction id " + finalCheckout.getReactionid() + " Not found" ));
-            Reaction checkinReaction =  reactionRepo.findReactionByReactionid(checkin.getReactionid()).orElseThrow(() -> new ResourceNotFoundException("reaction id " + finalCheckin.getReactionid() + " Not found" ));
+
+//            ClubsCheckInOutSummary finalCheckin = checkin;
+//            ClubsCheckInOutSummary finalCheckout = checkout;
```
Comment (L-Steinmacher) [inline src/main/java/com/lambdaschool/oktafoundation/controllers/LeaderboardController.java:68]: Will we be using these commented out items?

File: src/main/java/com/lambdaschool/oktafoundation/services/ClubServiceImpl.java [hunk 3]
```diff
@@ -53,7 +54,7 @@
     public Club findClubById(Long clubid) throws
             ResourceNotFoundException
     {
-        return clubrepos.findById(clubid)
+        return clubrepos.findClubByClubid(clubid)
```
Comment (L-Steinmacher) [inline src/main/java/com/lambdaschool/oktafoundation/services/ClubServiceImpl.java:57]: This makes it seem so much clearer. findById is vague.  Good call.

**Review by L-Steinmacher — APPROVED** [review]
Looks good.  Nice edits for clarity.

**Review by JackHenry15 — APPROVED** [review]
Looks good. Merging this

**Pull request merged**